pkg/kapp/app/app_filter.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import "time"
+
+type AppFilter struct {
+	CreatedAtBeforeTime *time.Time
+	CreatedAtAfterTime  *time.Time
+
+	Labels []string
+}
+
+func (f AppFilter) Apply(apps []App) ([]App, error) {
+	var result []App
+
+	for _, app := range apps {
+		lastChange, err := app.LastChange()
+		if err != nil {
+			return []App{}, err
```

Comment:
nit: Should we return `nil, err` instead of `[]App{}, err`.

Reply:
Done.